x-pack/legacy/plugins/canvas/types/functions.ts

```diff
@@ -25,6 +25,9 @@ export type UnionToIntersection<U> =
  */
 export type ValuesOf<T extends any[]> = T[number];
 
+type valueof<T> = T[keyof T];
+type ValuesOfUnion<T> = T extends any ? valueof<T> : never;
```

Comment:
nit: these would be good candidates for [@kbn/utility-types](https://github.com/elastic/kibana/tree/master/packages/kbn-utility-types) if they don't already exist there.